pkg/kubectl/cmd/run.go

```diff
@@ -196,7 +196,12 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 	}
 
 	params["env"] = cmdutil.GetFlagStringSlice(cmd, "env")
-
+	port := cmdutil.GetFlagInt(cmd, "port")
```

Comment:
this extra checks avoids deployment created successful but services created failed when --port=0 .
if adding these in generateService,the checks will not work, because createGeneratedObject is called front.
Can i remove the checks from generateService ?